src/cargo/lib.rs

```diff
@@ -41,6 +41,8 @@ use term::color::{BLACK};
 
 pub use util::{CargoError, CargoResult, CliError, CliResult, human, Config, ChainError};
 
+pub const CARGO_ENV: &'static str = "CARGO";
```

Comment:
I think something more verbose like `CARGO_EXECUTABLE` would be a tiny bit better, but we already use plain `RUSTC` and `RUSRDOC` variables for similar purposes, and in this regard just `CARGO` is indeed the best name here.